hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/OIDCTokenCredentialsProvider.java

```
@@ -0,0 +1,105 @@
+ * would be a backward-incompatible change.
+ *
+ * This credential provider must not fail in creation because that will
+ * break a chain of credential providers.
+ */
+@InterfaceAudience.Public
+@InterfaceStability.Stable
+public class OIDCTokenCredentialsProvider implements AWSCredentialsProvider {
+  public static final String NAME
+          = "org.apache.hadoop.fs.s3a.OIDCTokenCredentialsProvider";
+
+  /** Reuse the S3AFileSystem log. */
+  private static final Logger LOG = S3AFileSystem.LOG;
+
+  private String jwtPath;
+  private String roleARN;
+  private String sessionName;
+  private IOException lookupIOE;
+
+  public OIDCTokenCredentialsProvider(Configuration conf) {
+    try {
+      Configuration c = ProviderUtils.excludeIncompatibleCredentialProviders(
+                  conf, S3AFileSystem.class);
+      this.jwtPath = S3AUtils.lookupPassword(c, JWT_PATH, null);
+      this.roleARN = S3AUtils.lookupPassword(c, ASSUMED_ROLE_ARN, null);
+      this.sessionName = S3AUtils.lookupPassword(c, ASSUMED_ROLE_SESSION_NAME, null);
+    } catch (IOException e) {
+      lookupIOE = e;
+    }
+  }
+
+  public AWSCredentials getCredentials() {
+      if (lookupIOE != null) {
+          // propagate any initialization problem
+          throw new CredentialInitializationException(lookupIOE.toString(),
+                  lookupIOE);
+      }
+
+      LOG.debug("jwtPath {} roleARN {} sessionName {}", jwtPath, roleARN, sessionName);
+
+      if (!StringUtils.isEmpty(jwtPath) && !StringUtils.isEmpty(roleARN)) {
+          final AWSCredentialsProvider credentialsProvider =
+              WebIdentityTokenCredentialsProvider.builder()
+                  .webIdentityTokenFile(jwtPath)
+                  .roleArn(roleARN)
+                  .roleSessionName(sessionName)
+                  .build();
+          return credentialsProvider.getCredentials();
+      }
+      else throw new CredentialInitializationException(
```

better to throw explicit errors when the options are unset.